src/libs/actions/Subscription.ts

```diff
@@ -8,7 +14,50 @@ function openSubscriptionPage() {
     API.read(READ_COMMANDS.OPEN_SUBSCRIPTION_PAGE, null);
 }
 
-export {
-    // eslint-disable-next-line import/prefer-default-export
-    openSubscriptionPage,
-};
+function updateSubscriptionType(type: SubscriptionType) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+            value: {
+                type,
+                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                errors: null,
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+            value: {
+                type,
+                pendingAction: null,
+                errors: null,
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+            value: {
+                pendingAction: null,
+            },
```

Comment:
Sorry, just to expand on what I mentioned earlier - to restore to the existing value `type` value upon failure, we need to set the _existing_ value here. So you could either pass that into the function, or since there's only 2 options, do something like
```ts
type: type === CONST.SUBSCRIPTION.TYPE.ANNUAL ? CONST.SUBSCRIPTION.TYPE.PAYPERUSE : CONST.SUBSCRIPTION.TYPE.ANNUAL,
```

Current behavior if I force an error:

https://github.com/Expensify/App/assets/7277067/156a596c-6f3c-454d-a369-1f0e1ea1c1a8

Reply:
Ohh sorry I misread too!